beginner_source/dist_overview.rst

```diff
@@ -15,107 +15,62 @@ to the technology that can best serve your use case.
 Introduction
 ------------
 
-As of PyTorch v1.6.0, features in ``torch.distributed`` can be categorized into
-three main components:
-
-* `Distributed Data-Parallel Training <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
-  (DDP) is a widely adopted single-program multiple-data training paradigm. With
-  DDP, the model is replicated on every process, and every model replica will be
-  fed with a different set of input data samples. DDP takes care of gradient
-  communication to keep model replicas synchronized and overlaps it with the
-  gradient computations to speed up training.
-* `RPC-Based Distributed Training <https://pytorch.org/docs/stable/rpc.html>`__
-  (RPC) supports general training structures that cannot fit into
-  data-parallel training such as distributed pipeline parallelism, parameter
-  server paradigm, and combinations of DDP with other training paradigms. It
-  helps manage remote object lifetime and extends the
-  `autograd engine <https://pytorch.org/docs/stable/autograd.html>`__ beyond
-  machine boundaries.
-* `Collective Communication <https://pytorch.org/docs/stable/distributed.html>`__
-  (c10d) library supports sending tensors across processes within a group. It
+The PyTorch Distributed library includes a collective of parallelism modules,
+a communications layer (C10D), and infrastructure for launching and
+debugging large training jobs.
+
+These Parallelism Modules offer high-level functionality and compose with existing models:
+* `Distributed Data-Parallel (DDP) <https://pytorch.org/docs/stable/generated/torch.nn.parallel.DistributedDataParallel.html>`__
+* `Fully Sharded Data-Parallel Training (FSDP) <https://pytorch.org/docs/stable/fsdp.html>`__
+* TODO what to mention about FSDP2 vs FSDP
+* `Tensor Parallel <https://pytorch.org/docs/stable/distributed.tensor.parallel.html>`__
+* TODO link to pipeline parallel after stable docs link exists
+
+DTensor and DeviceMesh are primitives used to build parallelism in terms of sharded or replicated tensors on N-dimensional process groups.
+* `DTensor <https://github.com/pytorch/pytorch/blob/main/torch/distributed/_tensor/README.md>`__ represents a tensor that is sharded and/or replicated, and communicates automatically to reshard tensors as needed by operations.
+* `DeviceMesh <https://pytorch.org/docs/stable/distributed.html#devicemesh>`__ abstracts multiple dimensions of parallelism and manages the underlying ProcessGroups used for collective communications,
```

Comment:
Same here. Feels like it should be in backticks (``ProcessGroups ``) - will pass spellcheck after that.

Reply:
ok. i didnt see your comments till after i pushed a change to add words to spelling list. ProcessGroup is also a class.  Should i write it as ``ProcessGroup``s?  or where do I put the `s`?   Or do I just put it into the spelling list

Reply:
`ProcessGroup`s should be okay I think.

Reply:
hmm, the backticks worked for DTensor/DeviceMesh, but ProcessGroups rendered badly, like  \``ProcessGroup``s